design-system/react/src/components/ButtonLink/defs.ts

```diff
@@ -0,0 +1,29 @@
+import type { ButtonProps } from '../Button/defs';
+
+import type { Components } from '~/defs';
+import type { CreateComponent, HTMLProps } from '~/utils';
+
+type PropsToOmit =
+  | 'as'
+  | 'leftIcon'
```

Comment:
why omit icon-related props here?

Reply:
I omitted them because from what I know, the ButtonLink is more of a link than a button.
At least, that's what I picked from the examples.